uktrade/data-hub-frontend
# Migrate from `storybook-readme` to `@storybook/addon-docs`

## Description of change

The `storybook-readme` package has not been updated since 2020 and there have been several deprecation warnings and security alerts related to it. In order to replace it I have introduced Storybook's official addon for this purpose (`@storybook/addon-docs`).

Instead of installing the addon package directly, I have used the `@storybook/addon-essentials` package which contains the docs addon and also includes the action and control addons (as well as a few other things).

Annoyingly the addon doesn't work out of the box for `multiInstance` components, however I've come up with a workaround that I've outlined below.

This PR almost entirely consists of existing content/documentation that has been moved around. It was not possible to split this up into smaller PRs without causing inconsistencies within Storybook. The only new code that has been added are the various `addParameters({ component: ComponentName })` changes and a couple of components that didn't have PropTypes setup fully. Also the majority of the additions/deletions are changes to `package-lock`.

## Test instructions

Run `npm install` and `npm run storybook`. Go to any component and look at the 'Docs' page (components named in commits below will have documentation). You should be able to see the docs page.

## Screenshots
### Before

<img width="1468" alt="Screenshot 2022-07-29 at 16 54 02" src="https://user-images.githubusercontent.com/36161814/181797774-0d8f4e63-2d40-4ce5-8c7f-c94e50e38444.png">


### After

<img width="1454" alt="Screenshot 2022-07-29 at 16 54 10" src="https://user-images.githubusercontent.com/36161814/181797838-f47c99bc-d809-45d2-8a95-df5cb8a33c66.png">

## Checklist

[//]: # "When submitting a PR make sure the code review guidelines have been satisfied.
https://github.com/uktrade/data-hub-frontend/blob/master/docs/Code%20review%20guidelines.md"
